dynamic/clientset/dynamic_clientset.go

```diff
@@ -72,33 +81,27 @@ func (cs *Clientset) resource(apiResource *dynamicdiscovery.APIResource, namespa
 	if gv.Group != "" {
 		config.APIPath = "/apis"
 	}
-	dc, err := dynamic.NewClient(&config)
+	dc, err := dynamic.NewForConfig(&config)
 	if err != nil {
 		return nil, fmt.Errorf("can't create dynamic client for resource %v in apiVersion %v: %v", apiResource.Name, apiResource.APIVersion, err)
 	}
+	resourceGVR := getResourceGVR(apiResource)
 	return &ResourceClient{
-		ResourceInterface: dc.Resource(&apiResource.APIResource, namespace),
-		dc:                dc,
-		gv:                gv,
-		resource:          apiResource,
+		NamespaceableResourceInterface: dc.Resource(resourceGVR),
+		dc:                             dc,
+		gv:                             gv,
+		resource:                       apiResource,
 	}, nil
 }
 
 type ResourceClient struct {
-	dynamic.ResourceInterface
+	dynamic.NamespaceableResourceInterface
 
-	dc       *dynamic.Client
+	dc       dynamic.Interface
```

Comment:
We might not need to store `dc` here anymore, since we no longer need to make a new client just to change the namespace.

Reply:
Removal of `dc` means we have to find another way of initializing `ResourceClient.NamespaceableResourceInterface`. I will see what is the possible workaround. This will take me some time because I am basically have zero knowledge on API machinery.